### PR TITLE
Overhaul account/profile error handling

### PR DESCRIPTION
**Pull request by bhollis**
This PR overhauls our error handling around loading accounts and the profile, in order to avoid some common issues.

1. We no longer should flip to another account if the account you're used to fails to load. Instead, we'll show an error screen.
8. If the profile fails to load, we show a new error screen, instead of just showing the loading animation forever and a popup. If the inventory has loaded before and is now failing to load, we still show popups.
2. Accounts are always ordered by last-played-date in the menu, and if we have to default to an account, we default to the most recently played.
3. D1 accounts afflicted with the Twitch linking bug now display in the sidebar, and offer helpful advice if you try to view them.
4. Inline twitter feeds for both ThisIsDIM and BungieHelp on the top-level error screens.
5. Non-JSON responses from Bungie.net (like happened the other day) now show as "Bungie.net is having difficulties" instead of printing the JSON exception.
6. We now use the list of accounts loaded from IndexedDB without waiting to hear what Bungie.net says, and if Bungie.net errors we can keep going. Previously this saved data wasn't doing us much good. This also means the site generally loads faster.
7. Accounts loading has been generally Reduxified.
8. Last viewed platform is now stored in localStorage, not SyncService.
9. Error boundaries look better (like error notifications).

Screen Shot 2020-04-04 at 9 57 28 PM
Screen Shot 2020-04-04 at 9 43 07 PM

### CODE REVIEW

File: config/i18n.json [hunk 1]
```diff
@@ -9,7 +9,13 @@
...
     "Stadia": "Stadia",
     "Steam": "Steam",
-    "Xbox": "XB1"
+    "Xbox": "XB1",
+    "ErrorLoading": "Unable to load Destiny Accounts from Bungie.net",
+    "MissingTitle": "Account Not Found",
+    "MissingDescription": "The account you're trying to view is not an account linked to your Bungie.net profile.",
+    "UnlinkTwitch": "Linking your Twitch account to Bungie.net will cause your D1 accounts to stop working.",
+    "UnlinkTwitchButton": "Unlink Twitch",
+    "ErrorLoadInventory": "Unable to load Destiny profile"
```
Comment (sundevour) [inline config/i18n.json:18]:
```suggestion
    "ErrorLoadInventory": "Unable to load your Destiny profile"
```
"profile" is vague to intermediate level users

File: config/i18n.json [hunk 2]
```diff
@@ -138,6 +144,9 @@
   "ErrorBoundary": {
     "Title": "Something went wrong"
   },
+  "ErrorPanel": {
+    "Description": "This is likely an issue that affects all third-party apps - check to see if Bungie.net or the Companion App works."
```
Comment (sundevour) [inline config/i18n.json:148]:
```suggestion
    "Description": "This is likely an issue that affects all third-party apps -- check to see if Bungie.net loads, or the Destiny 2 Companion App works."
```

**Comment by bhollis** [comment]
I like the idea of a checklist. 

A while back I made it so we can proceed with a cached manifest even if the manifest API fails. But it’d be a good diagnostic.

**Comment by sundevour** [comment]
it could go a long way toward providing non-technical users with the basic reassurance that yes, DIM is working

**Comment by sundevour** [comment]
a literal message that ways DIM is working. the actual perfect feature.

**Comment by bhollis** [comment]
I’ve been pondering on this and I’m not actually sure how to do the checklist in a way that conveys the right info - there are so many different ways things could be broken. What I try to do in this PR is differentiate between a Bungie service error and a JS error at least. 

I’ll probably merge this so we have coverage for upcoming instability and keep thinking of ways to clarify.

**Pull request merged**